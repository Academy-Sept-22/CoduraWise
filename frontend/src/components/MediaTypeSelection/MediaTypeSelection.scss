.media-selection {
  width: 88%;
  margin: 10px;

  border: none;
  padding: 1em;

  font-family: monospace, sans-serif;
  font-size: 0.9em;

  & legend {
    font-size: 50em;
    margin: 50px;
  }
  & input {
    margin-left: 5px;
  }
  & label {
    margin-left: 5px;
  }
}
