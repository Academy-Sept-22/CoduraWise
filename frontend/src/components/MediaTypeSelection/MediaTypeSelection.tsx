import React from "react";
import { MediaTypes } from "../../models/MediaTypes"

const MediaTypeSelection = () => {
  return(
    <form className="media-selection">
      {MediaTypes.map((mediaType : string) =>(
        <input key={mediaType} type="checkbox" name={mediaType} id={`${mediaType}media`} value={mediaType}/>
      ))}
    </form>
  )
}

export default MediaTypeSelection;