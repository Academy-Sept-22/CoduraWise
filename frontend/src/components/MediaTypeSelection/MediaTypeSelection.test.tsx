import TestRenderer from "react-test-renderer";
import MediaTypeSelection from "./MediaTypeSelection";
import React from "react";

test("snapshot test: the html matches", () => {
  const card = TestRenderer.create(
    <MediaTypeSelection/>,
  );

  expect(card.toJSON()).toMatchSnapshot();
});
