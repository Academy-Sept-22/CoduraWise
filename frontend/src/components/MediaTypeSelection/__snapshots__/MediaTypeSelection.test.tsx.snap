// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: the html matches 1`] = `
<form
  className="media-selection"
>
  <h3>
    Filter by
  </h3>
  <div>
    <input
      checked={false}
      id="videomedia"
      name="video"
      onChange={[Function]}
      type="checkbox"
      value="video"
    />
    <label
      htmlFor="videomedia"
    >
      video
    </label>
  </div>
  <div>
    <input
      checked={false}
      id="articlemedia"
      name="article"
      onChange={[Function]}
      type="checkbox"
      value="article"
    />
    <label
      htmlFor="articlemedia"
    >
      article
    </label>
  </div>
  <div>
    <input
      checked={false}
      id="bookmedia"
      name="book"
      onChange={[Function]}
      type="checkbox"
      value="book"
    />
    <label
      htmlFor="bookmedia"
    >
      book
    </label>
  </div>
  <div>
    <input
      checked={false}
      id="tutorialmedia"
      name="tutorial"
      onChange={[Function]}
      type="checkbox"
      value="tutorial"
    />
    <label
      htmlFor="tutorialmedia"
    >
      tutorial
    </label>
  </div>
  <div>
    <input
      checked={false}
      id="coursemedia"
      name="course"
      onChange={[Function]}
      type="checkbox"
      value="course"
    />
    <label
      htmlFor="coursemedia"
    >
      course
    </label>
  </div>
</form>
`;
